internal/storage/storage_handle.go

```diff
@@ -309,12 +312,22 @@ func NewStorageHandle(ctx context.Context, clientConfig storageutil.StorageClien
 		if err != nil {
 			return nil, fmt.Errorf("error in getting clientOpts for gRPC client: %w", err)
 		}
-		rawStorageControlClient, err = storageutil.CreateGRPCControlClient(ctx, clientOpts)
+		rawStorageControlClientWithoutGaxRetries, err = storageutil.CreateGRPCControlClient(ctx, clientOpts, true)
+		if err != nil || rawStorageControlClientWithoutGaxRetries == nil {
```

Comment:
I think, nil error via `CreateGRPCControlClient` should ensure non-nil controlClient.  With that we don't need to check both.